### PR TITLE
Test migration script for court location

### PR DESCRIPTION
**Pull request by shankar-pankhania-hmcts**

### CODE REVIEW

**Review by CCHAPMAN01 — COMMENTED** [review]
When running these locally, the 2 data sets do not match - looks like the loc_attend_time  is not migrated correctly. Please ensure you run and validate your tests and raise bug tickets where appropriate to get them passing/reconciling.
I have raised a ticket for this issue: https://centralgovernmentcgi.atlassian.net/browse/JM-7463 - will let you know when it's fixed

File: Postgres old schema to new schema migration/test/0002 - court location/0002 - court location audit test verification.sql [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    revision int8 not null,
+    rev_type int4 null,
+    loc_code varchar(3) not null,
+    public_transport_soft_limit numeric(8, 5) null,
+    taxi_soft_limit numeric(8, 5) null);
+
+INSERT INTO temp_court_location_audit (revision,rev_type,loc_code,public_transport_soft_limit,taxi_soft_limit) VALUES
+(5,0,'417',4.00000,5.00000),
+(6,0,'440',5.00000,10.00000),
+(7,0,'442',20.00000,50.00000);
```
Comment (CCHAPMAN01) [inline Postgres old schema to new schema migration/test/0002 - court location/0002 - court location audit test verification.sql:11]: you will need to truncate the rev_info table in test set up and restart the identity column to guarantee you always get the same revision numbers when re-running the tests

File: Postgres old schema to new schema migration/test/0002 - court location/0002 - court location test verification.sql [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+        location_address,
+        region_id,
+        yield,
+        voters_lock,
+        term_of_service,
+        tdd_phone,
+        loc_signature,
+        public_transport_soft_limit,
+        cost_centre,
+        assembly_room,
```
Comment (CCHAPMAN01) [inline Postgres old schema to new schema migration/test/0002 - court location/0002 - court location test verification.sql:57]: assembly room gets migrated at a later date (as part of the courtroom migration script) so can be excluded from these checks (or be hard coded to expect null here)

File: Postgres old schema to new schema migration/test/0002 - court location/0002 - court location test verification.sql [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+    term_of_service varchar(20) null,
+    tdd_phone varchar(12) null,
+    loc_signature varchar(30) null,
+    public_transport_soft_limit numeric(8, 5) null,
+    cost_centre varchar(9) not null,
+    assembly_room int8 null,
+    taxi_soft_limit numeric(8, 5) null);
+
+insert into temp_court_location ("owner",loc_code,loc_name,loc_court_name,loc_attend_time,loc_address1,loc_address2,loc_address3,loc_address4,loc_address5,loc_address6,loc_zip,loc_phone,jury_officer_phone,location_address,region_id,yield,voters_lock,term_of_service,tdd_phone,loc_signature,public_transport_soft_limit,cost_centre,assembly_room,taxi_soft_limit) VALUES
+('417','417','COVENTRY','THE CROWN COURT AT COVENTRY','09:30:00','COVENTRY COMBINED COURT ','140 MUCH PARK STREET','COVENTRY',NULL,NULL,NULL,'CV1 2SN','01926 682425','01926 682 425','Coventry Crown Court,Coventry Combined Court,140 Much Park Street,Coventry CV1 2SN','MI',2.7,0,'Y','01926 682519','Jury Manager',NULL,'none',NULL,NULL),
```
Comment (CCHAPMAN01) [inline Postgres old schema to new schema migration/test/0002 - court location/0002 - court location test verification.sql:29]: need to insert the correct expected values for:

public_transport_soft_limit
cost_centre
taxi_soft_limit

File: Postgres old schema to new schema migration/test/0002 - court location/0002 - court location audit test verification.sql [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+   
+
+INSERT INTO temp_court_location_audit (revision,rev_type,loc_code,public_transport_soft_limit,taxi_soft_limit) VALUES
+ (16,0,'442',20.00000,50.00000),
+ (15,0,'440',5.00000,10.00000),
+ (14,0,'417',4.00000,5.00000);
+
+
+select
+        revision,
```
Comment (philhead1) [inline Postgres old schema to new schema migration/test/0002 - court location/0002 - court location audit test verification.sql:16]: can you remove the revision column as column should be autogenerated

File: Postgres old schema to new schema migration/test/0002 - court location/0002 - court location audit test verification.sql [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+        rev_type,
+        loc_code,
+        public_transport_soft_limit,
+        taxi_soft_limit
+from	temp_court_location_audit
+
+except
+
+select
+        revision,
```
Comment (philhead1) [inline Postgres old schema to new schema migration/test/0002 - court location/0002 - court location audit test verification.sql:26]: can you remove the revision column as column should be autogenerated

File: Postgres old schema to new schema migration/test/0002 - court location/0002 - court location audit test verification.sql [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+        rev_type,
+        loc_code,
+        public_transport_soft_limit,
+        taxi_soft_limit
+from	juror_mod.court_location_audit;
+
+
+select
+        revision,
```
Comment (philhead1) [inline Postgres old schema to new schema migration/test/0002 - court location/0002 - court location audit test verification.sql:35]: can you remove the revision column as column should be autogenerated

File: Postgres old schema to new schema migration/test/0002 - court location/0002 - court location audit test verification.sql [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+        rev_type,
+        loc_code,
+        public_transport_soft_limit,
+        taxi_soft_limit
+from	juror_mod.court_location_audit
+
+except
+
+select
+        revision,
```
Comment (philhead1) [inline Postgres old schema to new schema migration/test/0002 - court location/0002 - court location audit test verification.sql:45]: can you remove the revision column as column should be autogenerated

File: Postgres old schema to new schema migration/test/0002 - court location/0002 - court location audit test verification.sql [hunk 8]
```diff
@@ -0,0 +1,50 @@
+create temporary table if not exists temp_court_location_audit (
+    revision int8 not null,
```
Comment (philhead1) [inline Postgres old schema to new schema migration/test/0002 - court location/0002 - court location audit test verification.sql:2]: revision is not needed here so remove please

File: Postgres old schema to new schema migration/test/0002 - court location/0002 - court location audit test verification.sql [hunk 9]
```diff
@@ -0,0 +1,50 @@
+create temporary table if not exists temp_court_location_audit (
```
Comment (philhead1) [inline Postgres old schema to new schema migration/test/0002 - court location/0002 - court location audit test verification.sql:1]: can you insert the following above please?

drop table if exists temp_court_location_audit;

**Review by philhead1 — APPROVED** [review]

**Pull request merged**